src/Tickets/Commerce/Provider.php

```diff
@@ -100,6 +102,100 @@ public function register() {
 
 		// Cache invalidation.
 		add_filter( 'tec_cache_listener_save_post_types', [ $this, 'filter_cache_listener_save_post_types' ] );
+
+		add_action( 'tec_tickets_commerce_async_webhook_process', [ $this, 'process_async_stripe_webhook' ], 10, 5 );
+	}
+
+	/**
+	 * Process the async stripe webhook.
+	 *
+	 * @since TBD
+	 *
+	 * @param int    $order_id           The order ID.
+	 * @param string $new_status_wp_slug The new status WP slug.
+	 * @param array  $metadata           The metadata.
+	 * @param string $old_status_wp_slug The old status WP slug.
+	 * @param int    $current_try        The try.
+	 *
+	 * @throws Exception If the action fails after too many retries.
+	 */
+	public function process_async_stripe_webhook( int $order_id, string $new_status_wp_slug, array $metadata = [], string $old_status_wp_slug = '', int $current_try = 1 ): void {
+		$order = tec_tc_get_order( $order_id );
+
+		if ( ! $order || ! $order instanceof WP_Post || ! $order->ID ) {
+			return;
+		}
+
+		// The order is already there!
+		if ( $order->post_status === $new_status_wp_slug ) {
+			return;
+		}
```

Comment:
If the locking mechanism is working then we don't need to check it here again. The `can_apply_to` method already doing this same thing and is called by the `modify_status` method.

Reply:
The locking mechanism is working - this check aims to mark the action as succesfull instead of protecting as from something else.

If we didnt check this here, the action would fail during modify_status and as a result reschedule itself.